lib/image-stream/handlers.js

```diff
@@ -161,6 +163,17 @@ module.exports = {
         },
         transform: new PassThroughStream()
       };
+    }).catch((error) => {
+      if (/invalid footer/i.test(error.message)) {
+        throw errors.createUserError({
+          title: 'Invalid image',
+          description: `There was an error reading "${path.basename(file)}". `
+            + 'The image does not appear to be a valid Apple Disk Image, or may have the wrong filename extension.\n\n'
```

Comment:
Are we sure that any error messages containing `/invalid footer/i` will always be related to Apple Disk Images? Is this the same thing as `.dmg`? Would it be worth tweaking the message slightly to `... a valid Apple Disk Image (DMG), ...`? I'm not a Mac user, so I've got no idea how familiar people are likely to be with the terminology.